widgetssdk/src/main/java/com/glia/widgets/helper/ActivityWatcherForDialogs.kt

```diff
@@ -35,23 +46,20 @@ class ActivityWatcherForDialogs(
     @VisibleForTesting
     var resumedActivity: WeakReference<Activity?> = WeakReference(null)
 
-    fun init(dialogController: DialogController) {
-        app.registerActivityLifecycleCallbacks(this)
-        this.dialogController = dialogController
-    }
-
     override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {}
     override fun onActivityDestroyed(activity: Activity) {}
 
 
     override fun onActivityResumed(activity: Activity) {
         resumedActivity = WeakReference(activity)
```

Comment:
I think it is pointless here to keep the `activity` instance in `WeakReference` if, on the next line, you're passing it to another method. In this case, the `WeakReference` will have no effect. The `WeakReference`  must be like a holder of the activity instance.